code/modules/admin/verbs/adminpm.dm

```diff
@@ -185,3 +185,192 @@
+	set category = null
+	set name = "Admin PM Mob"
+	if(!holder)
+		to_chat(src, "<span class='warning'>Error: Admin-PM-Context: Only administrators may use this command.</span>")
+		return
+	if( !ismob(M) || !M.client )	return
+	cmd_admin_pm(M.client,null)
+	feedback_add_details("admin_verb","APMM") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+//shows a list of clients we could send PMs to, then forwards our choice to cmd_admin_pm
+/client/proc/cmd_admin_pm_panel()
+	set category = "Admin"
+	set name = "Admin PM"
+	if(!holder)
+		to_chat(src, "<span class='warning'>Error: Admin-PM-Panel: Only administrators may use this command.</span>")
+		return
+	var/list/client/targets[0]
+	for(var/client/T)
+		if(T.mob)
+			if(isnewplayer(T.mob))
+				targets["(New Player) - [T]"] = T
+			else if(isghost(T.mob))
+				targets["[T.mob.name](Ghost) - [T]"] = T
+			else
+				targets["[T.mob.real_name](as [T.mob.name]) - [T]"] = T
+		else
+			targets["(No Mob) - [T]"] = T
+	var/list/sorted = sortList(targets)
+	var/target = input(src,"To whom shall we send a message?","Admin PM",null) in sorted|null
+	cmd_admin_pm(targets[target],null)
+	feedback_add_details("admin_verb","APM") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+
+//takes input from cmd_admin_pm_context, cmd_admin_pm_panel or /client/Topic and sends them a PM.
+//Fetching a message if needed. src is the sender and C is the target client
+
+/client/proc/cmd_admin_pm(var/client/C, var/msg = null, var/datum/ticket/ticket = null)
+	if(prefs.muted & MUTE_ADMINHELP)
+		to_chat(src, "<span class='warning'>Error: Private-Message: You are unable to use PM-s (muted).</span>")
+		return
+
+	if(!istype(C,/client))
+		if(holder)	to_chat(src, "<span class='warning'>Error: Private-Message: Client not found.</span>")
+		else		to_chat(src, "<span class='warning'>Error: Private-Message: Client not found. They may have lost connection, so please be patient!</span>")
+		return
+
+	var/recieve_pm_type = "Player"
+	if(holder)
+		//mod PMs are maroon
+		//PMs sent from admins and mods display their rank
+		if(holder)
+			recieve_pm_type = holder.rank
+
+	else if(C && !C.holder)
+		to_chat(src, "<span class='warning'>Error: Admin-PM: Non-admin to non-admin PM communication is forbidden.</span>")
+		return
+
+	msg = sanitize(msg)
+
+	//get message text, limit it's length.and clean/escape html
+	if(!msg)
+		msg = input(src,"Message:", "Private message to [key_name(C, 0, holder ? 1 : 0)]") as text|null
+
+		if(!msg)	return
+		if(!C)
+			if(holder)	to_chat(src, "<span class='warning'>Error: Private-Message: Client not found.</span>")
+			else		to_chat(src, "<span class='warning'>Error: Private-Message: Client not found. They may have lost connection, so try using an adminhelp!</span>")
+			return
+
+		msg = sanitize(msg)
+
+	var/datum/client_lite/receiver_lite = client_repository.get_lite_client(C)
```

Comment:
we don't use client_lite much, you may be better off using /datum/player

Reply:
The old adminPM system used it too. This is why it's in there too.